docs/rototiller_task_reference.md

```diff
@@ -159,3 +162,112 @@ produces:
     command_name --switch --option opt arg arguments go last
 
     # what do you think ARG_OVERRIDE2 does?
+
+##Relationship between add_env and parameter name
+Any parameter or task can call the `add_env` method.  The context of the `add_env` method call and additional method calls determine its behavior.  The only location where `add_env` is not available is under an existing `add_env` method call.  
```

Comment:
this could use more on what the relationship is, and how it determine's its behavior.  or maybe just "see below".  but here might say something like "environment variable default values are different for tasks and command and its parameters"  "a task's environment variable does not have a parent with a useful name, so it does not set its default"